homeassistant/components/nuki/lock.py

```diff
@@ -2,42 +2,50 @@
 from abc import ABC, abstractmethod
 from datetime import timedelta
 import logging
+from typing import List
 
 from pynuki import NukiBridge
 from requests.exceptions import RequestException
 import voluptuous as vol
 
 from homeassistant.components.lock import PLATFORM_SCHEMA, SUPPORT_OPEN, LockEntity
-from homeassistant.const import CONF_HOST, CONF_PORT, CONF_TOKEN
+from homeassistant.const import CONF_HOST, CONF_PORT, CONF_TOKEN, CONF_WEBHOOK_ID
+from homeassistant.core import CALLBACK_TYPE
 from homeassistant.helpers import config_validation as cv, entity_platform
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from . import DOMAIN
+from .const import (
+    ATTR_BATTERY_CRITICAL,
+    ATTR_DATA,
+    ATTR_NUKI_ID,
+    ATTR_RING_ACTION_TIMESTAMP,
+    ATTR_UNLATCH,
+    DEFAULT_PORT,
+    DEFAULT_TIMEOUT,
+    ERROR_STATES,
+    EVENT_NUKI_BELL_RING,
+)
+from .webhook import register_webhook
 
 _LOGGER = logging.getLogger(__name__)
 
-DEFAULT_PORT = 8080
-DEFAULT_TIMEOUT = 20
-
-ATTR_BATTERY_CRITICAL = "battery_critical"
-ATTR_NUKI_ID = "nuki_id"
-ATTR_UNLATCH = "unlatch"
-
 MIN_TIME_BETWEEN_FORCED_SCANS = timedelta(seconds=5)
 MIN_TIME_BETWEEN_SCANS = timedelta(seconds=30)
 
-NUKI_DATA = "nuki"
-
-ERROR_STATES = (0, 254, 255)
-
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Required(CONF_HOST): cv.string,
         vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
         vol.Required(CONF_TOKEN): cv.string,
+        vol.Optional(CONF_WEBHOOK_ID): cv.string,
```

Comment:
We no longer accept additions or changes to the yaml config of integrations. See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

This integration needs to be refactored to use a config flow and config entries.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler

Reply:
And a refactor like that should be done separately, ie not in this PR.

Reply:
It seems like some of this work has been implemented in https://github.com/home-assistant/core/pull/41854 and https://github.com/home-assistant/core/pull/40933 -- it is not entirely clear if these PRs are still actively looked at or discarded, though.

Reply:
Those PRs look abandoned by the author. You're welcome to go ahead with a new PR to add config flow and config entry support.

Reply:
Yes, I've seen those PRs, but they look abandoned to me too.
I would probably start working on the config flow and config entry support in the next week or two, unless @SeraphimSerapis wants to work on it.

Reply:
I'm afraid I won't get to this earlier either. If you go ahead, @steinerl, I can see if I can be of help later down the road. Thanks for your work on this integration!

Reply:
Hey all, I haven't abandoned them, I was just unable to work on it the last couple of months, since my nuki isn't hooked up currently. I'll try to finalize them in the next couple of days so you can go further with your PR :)

Reply:
Hey @pree, do you think you'll find some time or shall we close them so someone else can attempt them?

Reply:
Hey @balloob sorry for the delay - just got my nuki installed again and did some work (including rebase) on the PR: https://github.com/home-assistant/core/pull/40933 . Sadly my new intercom is not supported by the opener so I'm unable to test if the opener integration still works, it was working properly before - maybe someone of you can test this. After this PR is merged I will finish my work on the config-flow PR ASAP.

Reply:
Hey @pree, the opener integration still works in my tests of #40933.
Thanks for your work!